SPECIFICATION.md

```diff
@@ -0,0 +1,320 @@
+By convention, the containerapp definition file is named `Nulecule`. The Nulecule is the primary file defining the containerapp and it's relationship to dependencies.
+
+### Data types 
+
+Common Name | `type`    | `format`    | Comments
+----------- | --------- | ----------- | --------------
+integer     | `integer` | `int32`     | signed 64 bits
+float       | `number`  | `float`     |
+string      | `string`  |             |
+byte        | `string`  | `byte`      |
+boolean     | `boolean` |             |
+date        | `string`  | `date`      | As defined by `full-date` - [RFC3339](http://xml2rfc.ietf.org/public/rfc/html/rfc3339.html#anchor14)
+dateTime    | `string`  | `date-time` | As defined by `date-time` - [RFC3339](http://xml2rfc.ietf.org/public/rfc/html/rfc3339.html#anchor14)
+password    | `string`  | `password`  | Used to hint UIs the input needs to be obscured.
+URL         | `URL`     | `URL`       | As defined by `URL` - [RFC3986 Section 1.1.3](https://tools.ietf.org/html/rfc3986#section-1.1.3)
+
+### Nulecule file schema
+
+#### Container Application Object
+This is the root object for the specification.
+
+Field Name  | Type                | Description
+----------  | :-----------------: | ------------
+id          | `string`            | **Required.** The machine readable id of the Container Application.
+specversion | `string`            | **Required.** The semantic version string of the Container Application Specification used to describe the app. The value MUST be `"0.0.2"` (current version of the spec).
+metadata    | `Metadata Object`   | **Optional** An object holding optional metadata related for the Container Application, this may include license information or human readable information.
+graph       | `Graph Object`      | **Required.** A list of depending containerapps. Strings may either match a local sub directory or another containerapp-spec compliant containerapp image that can be pulled via docker.
+requirements|`Requirements Object`| **Optional.** A list of requirements of this containerapp.
+
+
+#### Metadata Object
+
+Metadata for the Container Application.
+
+##### Fields
+
+Field Name     | Type            | Description
+---------------|:---------------:| ------------
+name           | `string`        | **Optional.** A human readable name of the containerapp.
+appversion     | `string`        | **Optional.** The semantic version string of the Container Application.
+description    | `string`        | **Optional.** A human readable description of the Container Application. This may contain information for the deployer of the containerapp.
+license        | `License Object`| **Optional.** The license information for the containerapp.
+arbitrary_data | `string`        | **Optional.** Arbitrary `key: value` pair(s) of metadata. May contain nested objects.
+
+##### Metadata Object Example
+
+```yaml
+metadata:
+  name: myapp
+  appversion: 1.0.0
+  description: description of myapp
+  foo: bar
+  othermetadata:
```

Comment:
How's ``othermetadata`` gonna be consumed? Or is it just a ``readonly`` info not used during application run/stop?